Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -97,10 +105,51 @@ private static InputActionAsset CreateNewActionAsset()
                     AssetDatabase.AddObjectToAsset(actionReference, asset);
                 }
             }
+        }
 
-            AssetDatabase.SaveAssets();
+        /// <summary>
+        /// Updates the input action references in the asset by updating names, removing dangling references
+        /// and adding new ones.
+        /// </summary>
+        /// <param name="asset"></param>
+        internal static void UpdateInputActionReferences()
+        {
+            var asset = GetOrCreate();
+            var existingReferences = InputActionImporter.LoadInputActionReferencesFromAsset(asset).ToList();
 
-            return asset;
+            // Check if referenced input action exists in the asset and remove the reference if it doesn't.
+            foreach (var actionReference in existingReferences)
+            {
+                var action = asset.FindAction(actionReference.action.id);
+                if (action == null)
+                {
+                    actionReference.Set(null);
+                    AssetDatabase.RemoveObjectFromAsset(actionReference);
+                }
+            }
+
+            // Check if all actions have a reference
+            foreach (var action in asset)
+            {
+                var actionReference = existingReferences.FirstOrDefault(r => r.m_ActionId == action.id.ToString());
```

Comment:
Yes. Since this is controlled by us in the InputManager.asset.
If there's an action, only one reference is created. We never create more than 1 reference for an action, at the moment.